### ISSUE TITLE
Che Theia Dependency Analytics plugin doesn't work

### ISSUE THREAD

**Issue by mmorhun** [eclipse-che/che#14942]
### Describe the bug
Invocation of `Dependency Analytics Report ...` on a `pom.xml` or `package.json` using Dependency Analytics plugin ends up in `Error: command 'extension.fabric8AnalyticsWidgetFullStack' not found`.

### Che version
multiuser, nightly

### Steps to reproduce

1. Create a nodejs or java workspace
2. Add `Dependency Analytics` plugin
3. Start the workspace
4. Open `package.json` or `pom.xml` file
5. From context menu run `Dependency Analytics Report ...`
6. After a minute the followind error appears in the browser dev-console:
```
Uncaught (in promise) Error: command 'extension.fabric8AnalyticsWidgetFullStack' not found
```

### Expected behavior
The report is shown to the user

### Runtime
minishift v1.34.1+c2ff9cb

### Screenshots
![dependency-analytics-report](https://user-images.githubusercontent.com/15607393/67211251-1b443b80-f423-11e9-88d2-88a86e6d3c42.png)

### Additional context
Probably something is broken in our Che Theia plugin API, because the plugin [registers the command](https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/blob/master/package.json#L69).

**Comment by mmorhun** [comment]
I've just checked - the command is invoked as expected in Eclipse Theia

**Comment by ibuziuk** [comment]
@invincibleJai @arajkumar could you please take a look

**Comment by mmorhun** [comment]
Yevhen and I did additional testing and it turned out that the root problem is in activation events. Due to some reasons the `activate` method wasn't called.

**Comment by arajkumar** [comment]
I did a quick test & it works in my minishift instance. I tried it with "next" channel,
<img width="1792" alt="Screenshot 2019-10-22 at 10 55 57 PM" src="https://user-images.githubusercontent.com/3874763/67312728-752a2b80-f51f-11e9-902f-78634f2fac90.png">

**Comment by mmorhun** [comment]
It sometimes works, sometimes not. If you have projects cloned - probably it will work. Better chance to reproduce the issue is to create a new workspace with some projects, which are cloned after workspace start.

**Comment by mmorhun** [comment]
Turned out that the presence of a project doesn't make it work in 100% cases...

**Comment by tsmaeder** [comment]
While the merged PR fixes the immediate problem, I think we should do some follow-up work:

1. We cannot trace remote method invocations: console.log, etc. are redirected, so logging messages leads to infinite loop
2. From the exception stack trace about the "onDidDeploy" it's completely unclear that this will break plugin container initialization, can we please add a try/catch covering the critical section?
3. The remote plugin initialization mechanism is undocumented. Can we please add a comment somewhere that describes what is happening? Which messages are sent? It's a non-obvious algorithm distributed over multiple files.
4. We use the string "internal" to mark special messages. Can we at least put that in a constant so that all users of that special tag can be found?

**Comment by mmorhun** [comment]
When I mostly investigate the problem on my local minishift instance I've also tried to do it on che.openshift.io and the bug is reproducible there too (Che 7.1 with unchanged images). Steps:
1. Create a workspace for node express application (or any other stack with node)
2. Replace sample project with [che-theia](https://github.com/eclipse/che-theia)
3. Open workspace
4. Run analyzation on the root package.json file. First time it started the command, but it then fails to analyze the dependencies. When try to rerun the command, the known error appears:
![openshift-analytics](https://user-images.githubusercontent.com/15607393/67555455-fcdb8a00-f719-11e9-9d57-46ef99a29468.png)
So, the bug might not be related to first start or not yet cloned sources...

**Comment by nickboldt** [comment]
Will this be included in 7.3.1? Or is it slipping because not done by EOD Oct 25?

**Comment by mmorhun** [comment]
After rebase and the PRs merge I cannot reproduce the issue any more and the analytics plugin works for me (it fails to analyze Che and Che-Theia projects, but this is issue of the plugin itself).
Unfortunately we didn't find the root cause of this issue. The problem didn't have reliable steps to reproduce and happened randomly. Also it was almost impossible to reproduce it in debug mode. Anyway, now the problem is gone, so I 'm closing this issue. If anyone is able to reproduce it again with the latest changes - feel free to reopen this issue.
